console/src/main/java/org/georchestra/console/ws/backoffice/delegations/DelegationController.java

```diff
@@ -0,0 +1,111 @@
+ *
+ * This file is part of geOrchestra.
+ *
+ * geOrchestra is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free
+ * Software Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * geOrchestra is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * geOrchestra.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.georchestra.console.ws.backoffice.delegations;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.georchestra.console.dao.DelegationDao;
+import org.georchestra.console.ds.DataServiceException;
+import org.georchestra.console.ds.RoleDao;
+import org.georchestra.console.model.DelegationEntry;
+import org.georchestra.lib.file.FileUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+@Controller
+public class DelegationController {
+
+	private static final Log LOG = LogFactory.getLog(DelegationController.class.getName());
+
+	private static final String BASE_MAPPING = "/private";
+	private static final String REQUEST_MAPPING = BASE_MAPPING + "/delegation";
+
+	@Autowired
+	private DelegationDao delegationDao;
+	@Autowired
+	private RoleDao roleDao;
+
+	@RequestMapping(value=REQUEST_MAPPING + "/delegations", method= RequestMethod.GET, produces = "application/json; charset=utf-8")
+	@ResponseBody
+	public String findAll() throws JSONException {
+
+		// TODO filter if request came from delegated admin
+		Iterable<DelegationEntry> delegations = this.delegationDao.findAll();
+
+		JSONArray res = new JSONArray();
+		for(DelegationEntry delegation: delegations)
+			res.put(delegation.toJSON());
+
+		return new JSONObject().put("delegations", res).toString();
+	}
+
+	@RequestMapping(value=REQUEST_MAPPING + "/{uid}", method= RequestMethod.GET, produces = "application/json; charset=utf-8")
+	@ResponseBody
+	public String findByUid(@PathVariable String uid) throws JSONException {
+
+		// TODO test if uid correspond to connected user if request came from delegated admin
+		DelegationEntry delegation = this.delegationDao.findOne(uid);
+
+		JSONObject res = new JSONObject();
+		res.put("orgs", delegation.getOrgs());
```

Comment:
also set `uid` field in returned object ?

Reply:
Done with 2f9b7bd